libopenarc/arc.c

```diff
@@ -1965,6 +2015,15 @@ arc_validate_msg(ARC_MESSAGE *msg, u_int setnum)
 	msg->arc_selector = arc_param_get(kvset, "s");
 	msg->arc_domain = arc_param_get(kvset, "d");
 
+	/* store algorithm in msg, needed for arc_get_key() */
+	alg = arc_param_get(kvset, "a");
+	status = arc_parse_algorithm(msg, alg, &nid);
+	if (status != ARC_STAT_OK)
+	{
+		arc_error(msg, "hash algorithm not correctly retrieved for algorithm %s", alg);
+		return ARC_STAT_BADALG;
```

Comment:
```suggestion
		return status;
```
I'm not sure it makes sense to call `arc_error()` here; a more specific error was already set by the called function, and I believe this will just overwrite that.

Reply:
Thanks @flowerysong good point, now implemented